### PR TITLE
parcel v1 to v2 migration

### PR DESCRIPTION
**Pull request by Himavanth**
## Description
**TODO**
- [x]  Parcel cleanup before exit

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Himavanth** [comment]
@moritzraho @purplecabbage @sandeep-paliwal @shazron

File: src/bundle.js [hunk 1]
```diff
@@ -49,22 +49,30 @@ module.exports = async (entryFile, dest, options = {}, log = () => {}) => {
...
   }
 
   aioLogger.debug(`bundle bundleOptions: ${JSON.stringify(parcelBundleOptions, null, 2)}`)
   log(`bundling ${entryFile}`)
-  const bundler = new Bundler(entryFile, parcelBundleOptions)
+  const bundler = new Bundler(parcelBundleOptions)
 
   const cleanup = async () => {
     aioLogger.debug('cleanup bundler...')
-    await bundler.stop()
```
Comment (sandeep-paliwal) [inline src/bundle.js:67]: I guess this needs to be replaced with onBuildEnd event from the bundler for the cleanup.

File: src/bundle.js [hunk 2]
```diff
@@ -49,22 +49,30 @@ module.exports = async (entryFile, dest, options = {}, log = () => {}) => {
...
-    outDir: dest,
-    contentHash: true, // false if dev, true if prod ??
-    watch: false, // currently false if dev true if prod,
-    minify: false,
-    logLevel: 1,
+    entries: entryFile,
+    defaultConfig: require.resolve('@parcel/config-default'),
+    shouldDisableCache: false,
+    targets: {
+      action: {
```
Comment (moritzraho) [inline src/bundle.js:56]: why `action` is it some parcel jargon or you choose the target key ?

**Review by moritzraho — COMMENTED** [review]
some first comments, code looks good to me but need to test

File: package.json [hunk 3]
```diff
@@ -39,7 +39,7 @@
     "klaw": "^3.0.0",
     "lodash.clonedeep": "^4.5.0",
     "mime-types": "^2.1.24",
-    "parcel-bundler": "1.12.3"
+    "parcel": "^2.0.0-beta.2"
```
Comment (moritzraho) [inline package.json:42]: is there an eta on release ? just wondering if we want to rely on a beta

Comment (Himavanth) [reply package.json:42]: Unfortunately no ETA. Just that it has been in beta for about a year now.

File: src/bundle.js (hunk 2 above)
Comment (Himavanth) [reply src/bundle.js:56]: It can be anything.

File: src/bundle.js (hunk 1 above)
Comment (Himavanth) [reply src/bundle.js:67]: For the onBuildEnd of Parcel V1, we have reporter plugin in V2. But i think it's different from the stop API. This event handler is to see what assets were built but not for cleanup.

File: src/bundle.js (hunk 2 above)
Comment (moritzraho) [reply src/bundle.js:56]: ok should we use web-assets key instead ?

**Comment by icaraps** [comment]
@Himavanth when trying to provide my custom parcel config with `.parcelrc`at the root of the project, I get a build failure
```
 Error: Invalid Parcel Config
```

```
{
  "extends": ["@parcel/config-default"],
  "reporters":  ["...", "parcel-reporter-static-files-copy"]
}
```

Do you know if I'm doing something wrong ? Or is it not possible anymore to provide a custom parcel config ?

**Comment by icaraps** [comment]
There is some mismatch in the latest tags of parcel and the associated npm packages. Parcel has beta.2 as latest but parcel/config-default has alpha.3. 
Using `npm i --save-dev @parcel/config-default@2.0.0-beta.3.1` I could get past the invalid parcel config thanks @Himavanth !

**Comment by icaraps** [comment]
@Himavanth  I noticed the `.parcel-cache` folder is not gitignored by default... I think it should WDYT ?

**Comment by Himavanth** [comment]
@icaraps good point. I will add it.

**Pull request merged**